File: use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/solver/justifications/StudentGroupSubjectVarietyJustification.java
package org.acme.schooltimetabling.solver.justifications;

import ai.timefold.solver.core.api.score.stream.ConstraintJustification;

import org.acme.schooltimetabling.domain.Lesson;

public record StudentGroupSubjectVarietyJustification(String studentGroup, Lesson lesson1, Lesson lesson2) implements ConstraintJustification {

}

File: getting-started/quarkus-integration-kotlin/src/main/kotlin/org/acme/kotlin/schooltimetabling/solver/justifications/StudentGroupSubjectVarietyJustification.kt
package org.acme.kotlin.schooltimetabling.solver.justifications

import ai.timefold.solver.core.api.score.stream.ConstraintJustification
import org.acme.kotlin.schooltimetabling.domain.Lesson

data class StudentGroupSubjectVarietyJustification(
    val studentGroup: String,
    val lesson1: Lesson,
    val lesson2: Lesson,
    val description: String
) :
    ConstraintJustification {

    constructor(
        studentGroup: String,
        lesson1: Lesson,
        lesson2: Lesson
    ) : this(
        studentGroup, lesson1, lesson2,
        "Student Group '%s' has two consecutive lessons on '%s' at '%s %s' and at '%s %s'"
            .format(
                studentGroup,
                lesson1.subject,
                lesson1.timeslot?.dayOfWeek,
                lesson1.timeslot?.startTime,
                lesson2.timeslot?.dayOfWeek,
                lesson2.timeslot?.startTime
            )
    )
}